### ISSUE TITLE
calling dataProvider callback with [], 0 results in warning.

### ISSUE THREAD

**Issue by davidmaxwaterman** [vaadin/vaadin-grid#1637]
I'm getting this warning:

```
The <vaadin-grid> needs the total number of items in order to display rows. Set the total number of items to the `size` property, or provide the total number of items in the second argument of the `dataProvider`’s `callback` call.
```

when I call the dataProvider callback with a size of '0'. While my app is starting up, there is no data...but vaadin-grid calls the data provider automatically.

It seems sensible to invoke the callback to say there's no data, and I don't expect it to complain when I do so. Saying the size of the data is '0' is not the same as not saying what the size of the data is.

Of course, when my data is actually ready, I call 'clearCache' to get it to ask again.

**Comment by johnthad** [comment]
FWIW, I assign the `grid.dataProvider` after I have the data so I know what to set for the size. The data is part of my element's property. When the `updated(changedProperties)` method sees that property has changed, it calls the method that assigns the `grid.dataProvider` (as well as some other UI stuff).

**Comment by davidmaxwaterman** [comment]
Hrm, something to think about, I guess.

NB, I don't assign the data provider in javascript - I do it in html via binding. I find that more natural, somehow.

Still, I don't much understand what the problem is with an empty list. Perhaps I can just not call the callback if there is no data.

**Comment by johnthad** [comment]
> NB, I don't assign the data provider in javascript - I do it in html via binding. I find that more natural, somehow.

Agreed. It feels natural _especially_ since that's how the example is written.

> Still, I don't much understand what the problem is with an empty list. Perhaps I can just not call the callback if there is no data.

Not calling the callback is what I do. When `update()` calls my `_showPage()` function, if there are no records, I just return. (As for why, I've not looked into the code. I know that before `<vaadin-grid>` when I tried creating each row as the data arrived from the API I had, the browser choked.)

**Comment by davidmaxwaterman** [comment]
Well, the plot thickens.

I wrapped my callback in an 'if' statement, so it doesn't call it if 'length' is zero.

However, I *still* get the warning message.

**Comment by davidmaxwaterman** [comment]
I notice that setting the 'size' property in html via an attribute, prevents the warning.

That behaviour is very counter-intuitive. I suspect this component has been designed assuming people are mainly controlling it via javascript, which is annoying for people who prefer doing so via html.
